[API Shield] Remove beta mention in API Discovery (#7833)
cloudflare/cloudflare-docs: content/api-shield/security/api-discovery.md

### Machine Learning-based Discovery

> Your API endpoints are discovered with both the Session Identifier-based Discovery as well as the Machine Learning-based Discovery.

Reviewer: ```suggestion
Your API endpoints are discovered with both the Session Identifier-based Discovery and the Machine Learning-based Discovery.
```
